maintainers/scripts/haskell/regenerate-transitive-broken-packages.sh

```diff
@@ -12,4 +12,4 @@ dont-distribute-packages:
 EOF
 
 echo "Regenerating list of transitive broken packages ..."
-echo -e $(nix-instantiate --eval --strict maintainers/scripts/haskell/transitive-broken-packages.nix) | sed 's/\"//' | LC_ALL=C.UTF-8 sort -i >> $config_file
+NIXPKGS_ALLOW_BROKEN= nix-instantiate --eval --strict --json maintainers/scripts/haskell/transitive-broken-packages.nix | jq -r . | LC_ALL=C.UTF-8 sort -i >> $config_file
```

Comment:
There are other env vars as well, we unset them in `mark-broken.sh` which also calls this script.

Honestly, I think we should rather pass `--option restrict-eval true -I .` to all Nix invocations in the haskell scripts folder which would solve the problem once and for all…

Reply:
Agreed that would make the most sense, but is hackage2nix affected by environment variables as well?

Reply:
I think we'd need to patch this call as well to be safe: https://github.com/NixOS/cabal2nix/blob/0d1a5a7d227683e9d099665f645a8618ec48ef94/distribution-nixpkgs/src/Distribution/Nixpkgs/PackageMap.hs#L47

Reply:
Although it probably doesn't matter in practice, probably a good idea to do!